services/slam/builtin/builtin.go

```diff
@@ -115,15 +113,7 @@ func RuntimeConfigValidation(svcConfig *AttrConfig, model string, logger golog.L
 			model, svcConfig.ConfigParams["mode"])
 	}
 
-	for _, directoryName := range [4]string{"", "data", "map", "config"} {
-		directoryPath := filepath.Join(svcConfig.DataDirectory, directoryName)
-		if _, err := os.Stat(directoryPath); os.IsNotExist(err) {
-			logger.Warnf("%v directory does not exist", directoryPath)
-			if err := os.Mkdir(directoryPath, os.ModePerm); err != nil {
-				return "", errors.Errorf("issue creating directory at %v: %v", directoryPath, err)
-			}
-		}
-	}
+	slamConfig.SetupDirectories(svcConfig.DataDirectory, logger)
 
 	if slamMode == slam.Rgbd || slamMode == slam.Mono {
```

Comment:
Should this be part of SetupDirectories as well, or will that be done in the orbslam module?